plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/project/classpath/ClasspathResolver.java

```diff
@@ -160,4 +159,22 @@ public void resolveClasspathEntries(List<ClasspathEntry> entries) {
   public Set<String> getProjects() {
     return projects;
   }
+
+  @Override
+  public void onProjectClasspathChanged(ProjectClasspathChangedEvent event) {
+    getEntriesFromServer(event.getProject())
+        .then(
+            entries -> {
+              eventBus.fireEvent(new ClasspathChangedEvent(event.getProject(), entries));
```

Comment:
Actually, no meaningful difference here..
The problem is that I cannot generate correct parameters for `ClasspathChangedEvent` in `ExecuteClientCommandProcessor.execute(ExecuteCommandParams)` (and I think it's not a good place to do that). So, I just notify `ClasspathResolver` on the fact of changed classpath by using `ProjectClasspathChangedEvent`, so it can correctly create and fill `ClasspathChangedEvent` strictly after all server-side modification is finished.
OK. I'll add some javadoc to `ProjectClasspathChangedEvent`.

Reply:
PR is rebased